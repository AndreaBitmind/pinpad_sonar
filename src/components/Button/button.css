.button {
  background: #fb0905;
  border: 1px solid rgb(255, 255, 255);
  border-radius: 15px;
  font-size: 2.5rem;
  color: white;
  cursor: pointer;
  padding: 25px;
  margin: 4px;
  margin-bottom: 4px;
  width: 70px;
  height: 70px;
  display: flex;
  align-items: center;
  font-weight: 700;
  font-family: sans-serif;
}

.button:hover {
  background-color: #f58d0a;
  transition: 0.5s;
}
