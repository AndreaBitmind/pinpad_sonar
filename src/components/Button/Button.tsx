import "./button.css";

interface ButtonProps {
  digit: number;
  handleChange: (number: number) => void;
}

const Button = ({ digit, handleChange }: ButtonProps): JSX.Element => {
  return (
    <button className="button" onClick={() => handleChange(digit)}>
      {digit}
    </button>
  );
};

export default Button;
