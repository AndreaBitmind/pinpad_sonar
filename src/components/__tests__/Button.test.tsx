import TestRenderer from "react-test-renderer";
import Button from "../Button/Button";

describe("Given a Button component", () => {
  describe("When it's rendered in the screen", () => {
    test("Then it should show the Button component", () => {
      const digitTest: number = 5;
      const handleChangeTest = jest.fn(jest.fn());

      const expectedButtonComponent = TestRenderer.create(
        <Button handleChange={handleChangeTest} digit={digitTest} />
      );

      expect(expectedButtonComponent).toMatchSnapshot();
    });
  });
});
