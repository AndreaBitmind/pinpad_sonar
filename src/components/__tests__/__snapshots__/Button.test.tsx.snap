// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Button component When it's rendered in the screen Then it should show the Button component 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  5
</button>
`;
